.cascade-code/Chapman.edu/_components/left_nav/left_nav.vtl

```diff
@@ -0,0 +1,153 @@
+## Formerly LeftNav_Velocity. Can be used with index and non-index pages.
+## New left nav format that works without a block
+
+#macro ( setVars )
+    ## Parent depends on whether page is an index or not
+    #set ( $isIndex = $currentPage.name == 'index' )
+    #if ( $isIndex )
+        #set ( $parent = $currentPage.parentFolder )
+        #set ( $parentDir = $parent.parentFolder )     
+    #else
+        #set ( $parentDir = $currentPage.parentFolder )
+    #end
+#end
+
+## Builds left navigation
+#macro ( buildLeftNav )
+    #buildLeftNavTitle()
+    <ul>
+    #set ( $siblings = $parentDir.children )
+    #buildLeftNavMenu($siblings)
+    </ul>
+#end
+
+## Builds the header for the left nav
+#macro ( buildLeftNavTitle )
+    #set ( $folderIndexPage = $_.locatePage("${parentDir.path}/index" ) )
+    
+    <div class="leftTitle">
+        <ul>
+            <li class="title">
+                <a href="${folderIndexPage.link}" target="_top" title="${_EscapeTool.xml($folderIndexPage.label)}" >
+                    ${_EscapeTool.xml($folderIndexPage.label)}
+                </a>
+            </li>
+        </ul>
+    </div>
+#end
+
+## Builds menu in the left nav
+#macro ( buildLeftNavMenu $siblings )
+    #foreach ( $sibling in $siblings )
+        #set ( $isVisible = $sibling.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+
+        ## Sibling is a single page
+        #if ( $sibling.assetType == 'page' && $isVisible && ($sibling.name != 'index') )
+            #buildSiblingPageListItem($sibling)
+
+        ## Sibling is a folder
+        #elseif ( $sibling.assetType == 'folder' && $isVisible )
+            #set ( $indexPage = $_.locatePage("${sibling.path}/index") )
+            #set ( $visibleIndex = $indexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            
+            ## Only if the index page is also visible
+            #if ( $visibleIndex )
+                #buildSiblingFolderListItem($sibling $indexPage)
+            #end
+        #end
+    #end
+#end
+
+## Builds a basic page list item that has no dropdown (no children) in the left nav
+#macro ( buildSiblingPageListItem $siblingPage )
+    #set ( $siblingPageTitle = ${_EscapeTool.xml($siblingPage.label)} )
+
+    #if ( $siblingPage.path == $currentPage.path )
+        <li class="active">
+    #else
+        <li>
+    #end
+        <a href="${siblingPage.link}" title="$siblingPageTitle" class="s-p">$siblingPageTitle</a>
+    </li>
+#end
```

Comment:
I'm going to nitpick here again to press the idea again of isolating data from output/templating:

```
#macro ( buildSiblingPageListItem $siblingPage )
    ## Data
    #set ( $siblingPageTitle = ${_EscapeTool.xml($siblingPage.label)} )
    #if ( $siblingPage.path == $currentPage.path )
      #set( $liClass = "active" )
    #else
      #set( $liClass = "unstyled-inactive" )
    #end

    ## Output
    <li class="${liClass}">
        <a href="${siblingPage.link}" title="$siblingPageTitle" class="s-p">$siblingPageTitle</a>
    </li>
#end
```

Reply:
Can we leave the class name blank for list items that aren't active or do we have to give it some value like `unstyled-inactive`?

Reply:
It triggers my OCD but we can. 😄 

...I think. The question is: what does the WAVE tool say?

Reply:
WAVE doesn't get mad about it, but I'm fine with adding it

Reply:
If WAVE's ok with it, I am, too. Thanks for checking.